futures/src/lib.rs

```diff
@@ -95,7 +96,11 @@
 //!       // connect() returns a future of an AMQP Client
 //!       // that resolves once the handshake is done
 //!       lapin::client::Client::connect(stream, &ConnectionOptions::default())
-//!     }).and_then(|client| {
+//!    }).and_then(|(client, heartbeat_future_fn)| {
```

Comment:
We need to put the heartbeat in a separate thread to deal with issues like #54 where the "main" thread could be in a state where it does nothing until it gets data from stdin or such.